### PR TITLE
LUCENE-10603: Remove SSDV#NO_MORE_ORDS definition

### PR DESCRIPTION
**Pull request by gsmiller**
### Description (or a Jira issue link if you have one)

This is the last bit of work needed in LUCENE-10603 to actually remove the definition of `SSDV#NO_MORE_ORDS` and stop returning it from `SSDV#nextOrd()` implementations.

Note this will release with 10.0 and will not be backported to 9.x.

### CODE REVIEW

File: lucene/test-framework/src/java/org/apache/lucene/tests/index/AssertingLeafReader.java [hunk 1]
```diff
@@ -1055,12 +1051,9 @@ public long cost() {
     @Override
     public long nextOrd() throws IOException {
       assertThread("Sorted set doc values", creationThread);
-      assert lastOrd != NO_MORE_ORDS;
       assert exists;
       long ord = in.nextOrd();
       assert ord < valueCount;
-      assert ord == NO_MORE_ORDS || ord > lastOrd;
-      lastOrd = ord;
       return ord;
```
Comment (jpountz) [inline lucene/test-framework/src/java/org/apache/lucene/tests/index/AssertingLeafReader.java:1057]: Maybe we could also verify here that the caller is not calling `nextOrd` more than `count` times?

**Review by jpountz — APPROVED** [review]
Thanks for the cleanup!

File: lucene/test-framework/src/java/org/apache/lucene/tests/index/AssertingLeafReader.java (hunk 1 above)
Comment (gsmiller) [reply lucene/test-framework/src/java/org/apache/lucene/tests/index/AssertingLeafReader.java:1063]: Good idea. Will do.

File: lucene/core/src/java/org/apache/lucene/index/SortedSetDocValuesWriter.java [hunk 2]
```diff
@@ -391,11 +386,7 @@ public boolean advanceExact(int target) throws IOException {
 
     @Override
     public long nextOrd() {
-      if (limit == ordUpto) {
-        return NO_MORE_ORDS;
-      } else {
-        return ords.ords.get(ordUpto++);
-      }
+      return ords.ords.get(ordUpto++);
```
Comment (LuXugang) [inline lucene/core/src/java/org/apache/lucene/index/SortedSetDocValuesWriter.java:389]: Hi @gsmiller ,I saw `limit` was removed, did you mean Caller should take the responsibility to check whether `nextOrd()` is out of bound `docValueCount()`?

In `SortingSortedNumericDocValues`,  there is also a `limit` used to throw exception if `nextValue()` is out of bound. should we keep the same logic between them?

Comment (gsmiller) [reply lucene/core/src/java/org/apache/lucene/index/SortedSetDocValuesWriter.java:389]: @LuXugang yeah, that's exactly right. The caller is now responsible for using `docValueCount()` to determine how many values the positioned doc has, and shouldn't call `nextOrd()` more than that many times. You're right that we could remove `limit` from `SortingSortedNumericDocValues` as well. I just missed it. Would you like to follow up with a change to remove it? I can get to it next week too. Thanks for pointing this out!

**Pull request merged**